ocaml/database/metastore/config_field.mli

```diff
@@ -0,0 +1,93 @@
+(** Types used for [etcd] configuration fields.
+
+    The serialization here matches what etcd expects.
+ *)
+
+(** a [result] type with errors having [`Msg of string] type *)
+type nonrec ('a, 'b) result = ('a, ([> `Msg of string] as 'b)) result
+
+val ( let+ ) : ('a, 'b) result -> ('a -> ('c, 'b) result) -> ('c, 'b) result
+(** [let+ ok = opt in ...] executes the body when [opt] is [Some ok] with [ok] in scope.
+
+    This is the same as [Result.bind opt @@ fun ok -> ...]
+ *)
+
+(** an integer that is always > 0 *)
+type positive = private int64
+
+val typ_of_positive : positive Rpc.Types.typ
+(** [typ_of_positive] serializes the same as an [int64] *)
+
+val positive : int64 -> (positive, _) result
+(** [positive i] is [i] when [i > 0] or an error otherwise.  *)
+
+(** an integer that is always >= 0, where 0 means unlimited *)
+type limit = private int64
+
+val typ_of_limit : limit Rpc.Types.typ
+(** [typ_of_limit] serializes the same as an [int64] *)
+
+val positive_or_unlimited : int64 option -> (limit, _) result
+(** [positive_or_unlimited limit] declares an etcd style limit.
+
+      @param limit when [None] means unlimited (encoded as 0), otherwise [Some p], where [p > 0].
+
+      [0] is not accepted, because it means unlimited, and explicitly raises an
+  exception to avoid accidentally interpreting a "no values permitted" as
+  "unlimited values are permitted".
+   *)
+
+(** milliseconds, always > 0 *)
+type milliseconds = private int64
```

Comment:
They are intervals. The internals representation needs to match what etcd uses (so milliseconds as integers). We *can* chose to use a different type at the high level though, in the 'constructor' that creates 'milliseconds' types and it might make sense for that to take an 'Mtime.Span.t' (might also avoid confusion between seconds and milliseconds).